src/xwayland-unmanaged.c

```diff
@@ -209,9 +227,12 @@ handle_request_activate(struct wl_listener *listener, void *data)
 
 	/*
 	 * Validate that the unmanaged surface trying to grab focus is actually
-	 * a child of the topmost mapped view before granting the request.
+	 * a child of the active view before granting the request.
+	 *
+	 * FIXME: this logic appears to always allow activating an unmanaged
+	 * surface if an xdg-shell view is active - which seems inconsistent.
 	 */
-	struct view *view = desktop_topmost_focusable_view(server);
+	struct view *view = server->active_view;
 	if (view && view->type == LAB_XWAYLAND_VIEW) {
```

Comment:
Looks like that could be fixed with another branch like
```c
if (view && view->type == LAB_XWAYLAND_VIEW) {
	[..]
} else if (view) {
	return;
}
seat_focus_surface(seat, xsurface->surface);
```

Reply:
I think there are still inconsistencies even with that. It's a bit of a can of worms which is why I didn't want to do more in this PR than add a FIXME comment.

Reply:
In that case I suggest specifying the weird edge cases in the comment so we don't end up forgetting about them.

Reply:
Added a link to the discussion